presto-hive/src/main/java/com/facebook/presto/hive/RecordFileWriter.java

```diff
@@ -161,6 +163,8 @@ public void rollback()
     {
         try {
             recordWriter.close(true);
+            // perform explicit delete of written file here as recordWriter.close() ignore the abort flag.
+            path.getFileSystem(conf).delete(path, false);
         }
         catch (IOException e) {
```

Comment:
I think throwing an exception here is wrong, since `rollback()` is called in a loop in `HivePageSink.doAbort()` to rollback all the writers. This means we can leak file descriptors or other resources.